craft_cli/pytest_plugin.py

```diff
@@ -158,6 +158,19 @@ def assert_verbose(
         """
         return self._check(expected_text, "verbose", regex)
 
+    def assert_warning(
```

Comment:
Please add a test for this in [test_pytest_plugin.py](https://github.com/canonical/craft-cli/blob/main/tests/test_pytest_plugin.py)